src/plugins/intel_gpu/src/kernel_selector/cl_kernels/permute_f_y_axes.cl

```diff
@@ -0,0 +1,200 @@
+    const int y_idx = get_global_id(1);
+
+    __attribute__((opencl_unroll_hint(J_TIMES)))
+    for (int j = 0; j < J_TIMES; ++j) {
+        const int x_idx = x_start + j * VEC_SIZE;
+        IN_VEC_TYPE res = READ_VEC(0, &input[INPUT0_GET_INDEX(b_idx, f_idx, y_idx, x_idx)]);
+#if HAS_FUSED_OPS
+        FUSED_OPS_VEC;
+        OUT_VEC_TYPE result = FUSED_OPS_RESULT_VEC;
+#else
+        OUT_VEC_TYPE result = ACTIVATION(res, ACTIVATION_PARAMS);
+#endif
+        const int output_idx = OUTPUT_GET_INDEX(b_idx, y_idx, f_idx, x_idx);
+        WRITE_VEC(result, 0, &output[output_idx]);
+    }
+}
+
+#elif defined (THREE_DIM_TRANSPOSE)
+
+#ifdef SUB_GROUP_SIZE
+REQD_SUB_GROUP_SIZE(SUB_GROUP_SIZE)
+__attribute__((reqd_work_group_size(1, 1, SUB_GROUP_SIZE)))
+#endif
+KERNEL (permute_f_y_axes)(
+    const __global INPUT0_TYPE* input,
+    __global OUTPUT_TYPE* output
+#if HAS_FUSED_OPS_DECLS
+    , FUSED_OPS_DECLS
+#endif
+    )
+{
+    __local OUTPUT_TYPE transpose_buf[FEATURE_BLOCK_SIZE][FEATURE_BLOCK_SIZE][TILE_SIZE];
+
+    const int bf = get_global_id(2);
+    const int f_idx = bf % INPUT0_FEATURE_NUM;
+    const int b_idx = bf / INPUT0_FEATURE_NUM;
+
+    int bf_local = get_local_id(2);
+    int y_local = get_local_id(1);
+
+    const int x_begin = get_global_id(0) * TILE_SIZE;
+    const int y_begin = get_global_id(1) * FEATURE_BLOCK_SIZE;
+    const int f_begin = get_local_size(2) * get_group_id(2);
+
+    __attribute__((opencl_unroll_hint(FEATURE_BLOCK_SIZE)))
+    for (int j = 0; j < FEATURE_BLOCK_SIZE; ++j) {
+        __attribute__((opencl_unroll_hint(TILE_SIZE)))
+        for (int i = 0; i < TILE_SIZE; ++i) {
+            const int x_idx = x_begin + i;
+            const int y_idx = y_begin + j;
+#ifdef SUB_GROUP_SIZE
+            const uint input_offset = INPUT0_GET_INDEX(b_idx, f_idx, y_idx, x_idx) - get_sub_group_local_id();
+            INPUT0_TYPE res = DT_INPUT_BLOCK_READ(input, input_offset);
+#else
+            INPUT0_TYPE res = input[INPUT0_GET_INDEX(b_idx, f_idx, y_idx, x_idx)];
+#endif
+    #if HAS_FUSED_OPS
+            FUSED_OPS;
+            transpose_buf[bf_local][j][i]  = FUSED_OPS_RESULT;
+    #else
+            transpose_buf[bf_local][j][i] = ACTIVATION(res, ACTIVATION_PARAMS);
+    #endif
+        }
+    }
+
+    __attribute__((opencl_unroll_hint(FEATURE_BLOCK_SIZE)))
+    for (int j = 0; j < FEATURE_BLOCK_SIZE; ++j) {
+        __attribute__((opencl_unroll_hint(TILE_SIZE)))
+        for (int i = 0; i < TILE_SIZE; ++i) {
+            const int x_idx = x_begin + i;
+            const int f = f_begin + j;
+            const int y_idx = y_begin + bf_local;
+#ifdef SUB_GROUP_SIZE
+            const uint output_offset = OUTPUT_GET_INDEX(b_idx, y_idx, f, x_idx) - get_sub_group_local_id();
+            DT_OUTPUT_BLOCK_WRITE(output, output_offset, transpose_buf[j][bf_local][i]);
+#else
+            const int output_idx = OUTPUT_GET_INDEX(b_idx, y_idx, f, x_idx);
+            output[output_idx] = transpose_buf[j][bf_local][i];
```

Comment:
Can this pass benefit from this kernel? 
(No subgroup read, no vectorize). If not, how about get rid of this impl and prevent such case from the validation ?

Reply:
I also was thinking about it. It won't benefit, so I changed the "Validate" method for this case.